portfolio/src/main/webapp/script.js

```diff
@@ -15,95 +15,174 @@
+/** 
+ * Retrieve the url for the posting the comments section form and store it
+ * @async
+ * @returns {String} containing the url where to send the form
+ */
 async function fetchBlobUrl() {
-  const request = await fetch("/blobstore-upload-url");
-  // TODO get the relative link
+  const request = await fetch('/blobstore-upload-url');
   return await request.text();
 }
 
-/** Set the action attribute value in the comments' form */
-async function setActionAttr() {
-  const commentsForm = document.querySelector("#comment-form");
+/** 
+ * Retrieve user comments and display them
+ * @async
+ * @param {number} numComments the maximum number of comments that the user wants to display
+ * @var {number} MAX_COMMENTS the default number of comments that will be displayed if the user does not choose a number
+ */
+async function getUserComments(numComments = MAX_COMMENTS) {
+  try {
+    const response = await fetch(`/data?numComments=${numComments}`);
+    const data = await response.json();
 
-  if(commentsForm){
-    commentsForm.action = fetchBlobUrl();
+    const displayUserCommentsElement = document.getElementById('user-comments');
+    /** check to make sure that there is @var displayUserCommentsElement on the current page before displaying the comments*/
+    if (displayUserCommentsElement) {
+      displayUserCommentsElement.innerText = '';
+
+      for (let comment in data) {
+        displayUserCommentsElement.appendChild(createParagraphElement(`${data[comment].name}: ${data[comment].comment}`));
+        displayUserCommentsElement.appendChild(createImgElement(data[comment].imageUrl));
+      }
+    }
+  } catch (error) {
+    console.error('There was an error loading comments: ', error);
   }
 }
 
-/** Creates an <p> element containing comments. */
-function createElement(comment) {
-  const pElement = document.createElement('p');
-  pElement.innerHTML = `${comment.name}: ${comment.comment}`;
-  return pElement;
+/**
+ * Generate a random greeting
+ * @returns {String} a greeting from the list
+ */
+function generateRandomGreeting() {
+  //TODO use google translated to get greetings in different languages
+  const greetings =
+    ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+
+  // Pick a random greeting.
+  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  return greeting;
 }
 
-/** create img element to display the image submitted by user */
-function createImgElement(comment){
-  const imgElement = document.createElement('IMG');
-  imgElement.setAttribute("src", `${comment.imageUrl}`);
-  imgElement.setAttribute("alt", "Image uploaded by user");
-  return imgElement;
+/**
+ * Get the user's maximum if they type the number in the input field
+ */
+function updateNumComments() {
+  const userNumComments = document.getElementById('numComments').value;
+  if (userNumComments) {
+    getUserComments(userNumComments);
+  } else {
+    getUserComments();
```

Comment:
I think you can just call getUserComments(userNumComments) in both cases, no? 
If userNumComments = null or undefined, your getUserComments function definition will assign the default

Another question - what happens when userNumComments is 0?

Reply:
> I think you can just call getUserComments(userNumComments) in both cases, no?
> If userNumComments = null or undefined, your getUserComments function definition will assign the default
> 
> Another question - what happens when userNumComments is 0?

There won't be any comments shown if the user selects zero

Reply:
I'm not sure this is working correctly. If userNumComments is set to 0, `if (0)` will evaluate to false, meaning you will call `getUserComments()` with no param, which sets the num of comments to the default, correct?

Reply:
> I'm not sure this is working correctly. If userNumComments is set to 0, `if (0)` will evaluate to false, meaning you will call `getUserComments()` with no param, which sets the num of comments to the default, correct?
> 
> IIUC, you should be able to call `getUserComments(userNumComments)` in both cases, and if userNumComments is equal to undefined, then the default will still be triggered. Caveat: if userNumComments is equal to null, the approach won't work, so you have to double check to see what `element.getValue` returns when it is empty

I see what would make it better. I am supposed to check if the document with that ID exists and update it's value. I might have made an error, I will correct that now